akka-actor/src/main/scala-2.13+/akka/util/ByteString.scala

```diff
@@ -386,6 +393,15 @@ object ByteString {
       }
     }
 
+    override def copyToArray[B >: Byte](dest: Array[B], start: Int, len: Int): Int = {
+      // min of the bytes available to copy, bytes there is room for in dest and the requested number of bytes
+      val copied = math.max(math.min(math.min(len, length), dest.length - start), 0)
+      if (copied > 0) {
+        System.arraycopy(bytes, 0, dest, start, copied)
+      }
+      copied
```

Comment:
```suggestion
      0
```

Reply:
If it wasn't 0 that'd be incorrect

Reply:
Oops, I hallucinated an 'else' there. Turning this into a an if/else makes the change less appealing indeed. Probably fine to keep it as it is now :+1: .